### ISSUE TITLE
`*args` and `**kwargs` are optional arguments, but Pyright thinks differently when they are annotated as `ParamSpec.args` and `ParamSpec.kwargs`.

### ISSUE THREAD

**Issue by MapleCCC** [microsoft/pyright#2200]
**To Reproduce**
```python
from collections.abc import Callable
from typing_extensions import ParamSpec

P = ParamSpec("P")

def repr_func_call(func: Callable[P, object], *args: P.args, **kwargs: P.kwargs) -> str:

    arg_reprs = [repr(arg) for arg in args]
    arg_reprs.extend(k + "=" + repr(v) for k, v in kwargs.items())

    return func.__name__ + "(" + ", ".join(arg_reprs) + ")"

repr_func_call(print)
```

Pyright complains that ` Arguments missing for parameters "values", "sep", "end", "file", "flush"`.

**Expected behavior**
There should be no error emitted for the above code.

**VS Code extension or command-line**
Pyright command-line (built from the latest commit fc77cae)

**Comment by erictraut** [comment]
Thanks for the bug report. The problem here is not that `*args` and `**kwargs` are optional arguments, but rather that the function signature of `print` includes `*args` and parameters that have default values and can therefore be left unspecified by the caller. I've updated the logic that checks for missing parameters in a ParamSpec to exclude those cases. This will be fixed in the next release.

**Comment by MapleCCC** [comment]
Dear Eric, thanks for the quick fix. I tried it and it worked well.

While we are here, I have another question that might be a bug depending on, would it be considered a type error to report if we assign a variable of Any or Unknow type to args of type P.args where P is a ParamSpec ?

**Comment by erictraut** [comment]
Could you provide a concrete example? Do you mean something like this?

```python
def add_values(a: int, b: int):
    ...

def foo(x: Any, y: Any):
    repr_func_call(add_values, x, y) # No error
    repr_func_call(add_values, x) # Error: not enough parameters
    repr_func_call(add_values, x, y, y) # Error: too many parameters

    repr_func_call(add_values, a=x, b=y) # No error
    repr_func_call(add_values, a=x) # Error: not enough parameters
```

The type `Any` is a subtype (and a supertype) of all other types.

**Comment by MapleCCC** [comment]
A concrete example would be like this:

```python
def add_values(a: int, b: int):
    ...

def foo(*args, **kwargs):
    repr_func_call(add_values, *args, **kwargs)  # returns a string "add_values(1, 1)"

foo(1, 1)
```

Pyright complains that `Argument of type "Unknown" cannot be assigned to parameter "args" of type "P.args" in function "repr_func_call"`.

**Comment by erictraut** [comment]
Yes, it shouldn't emit an error in this case. I just pushed a fix for this.

**Comment by MapleCCC** [comment]
I try the fix and it is good. Thank you for your patience and the quick fix. I will close this issue then.

**Comment by erictraut** [comment]
Thanks for the confirmation. I'm going to leave the bug report open until after we publish the next release.

**Comment by erictraut** [comment]
This is included in Pyright 1.1.163, which I just published. It will also be in the next release of Pylance.
